source/base/timer.cc

```diff
@@ -1042,8 +1042,51 @@ TimerOutput::reset()
   timer_all.restart();
 }
 
+
+
+namespace
+{
+  /**
+   * Abort, should there be an exception being processed (see the error
+   * message).
+   */
+  void
+  check_exception(const std::string &scope_name)
```

Comment:
I know that the place where you copied this from calls this `check_exception`, but would you mind renaming this function `abort_on_pending_exception`?